org.eclipse.m2e.launching/src/org/eclipse/m2e/internal/launch/MavenConsoleLineTracker.java

```diff
@@ -212,14 +222,27 @@ private String getText(IRegion lineRegion) throws BadLocationException {
     return line0;
   }
 
-  private String getLineText(IRegion lineRegion) throws BadLocationException {
-    return console.getDocument().get(lineRegion.getOffset(), lineRegion.getLength()).strip();
+  private String getLineText(IRegion region, List<int[]> removedLocations) throws BadLocationException {
```

Comment:
@mihnita does the console provide a better (built in) way to obtain the unescaped text?

Reply:
> @mihnita does the console provide a better (built in) way to obtain the unescaped text?
> Or should we add that to the console, maybe in a more efficient way?

Nothing public.
The only place where that is needed is to put text in clipboard without codes.
And that is a simple regexp replace in [`AnsiConsoleUtils`](https://github.com/mihnita/eclipse.platform.debug/blob/master/org.eclipse.ui.console/src/org/eclipse/ui/internal/console/ansi/AnsiConsoleUtils.java)
```
AnsiConsoleUtils.ESCAPE_SEQUENCE_REGEX_TXT.matcher(plainText).replaceAll("");
```

Reply:
BTW, I know why the external maven uses colors on the level and the embedded one does not.

`org.eclipse.m2e.maven.runtime` requires `org.eclipse.m2e.maven.runtime.slf4j.simple`, which contains (in `jars` the simple slf4j logger (`slf4j-simple-*.jar`)

That logger does not output colors.

The external maven has a `maven-slf4j-provider-*.jar` in the `libs` folder that implements what's needed to behave like a slf4j provider.
And that provider produces color output for the levels.

What we see colored right now with the embedded maven are messages logged.
Because if I do `logger.warning("Hello \033[91m RED \033[m world!");` the message itself has colors.

---
Some options:
1. leave it as is
2. remove `slf4j-simple`, take `maven-slf4j-provider`, and we will behave 100% like the external maven
3. replace `slf4j-simple` with logback, which allows for colors and a lot of customization.
4. replace `slf4j-simple` with our own slf4j-provider, where we have full control and we know that nobody will change it behind our back

I played with option 3, and worked without problems.
And I don't expect trouble with option 2 either.
Option 4 would be more work.

Reply:
I've also checked option 2, using `maven-slf4j-provider`
Works, and the output looks like the one from external maven.

Reply:
> > @mihnita does the console provide a better (built in) way to obtain the unescaped text?
> > Or should we add that to the console, maybe in a more efficient way?
> 
> Nothing public. The only place where that is needed is to put text in clipboard without codes. And that is a simple regexp replace in [`AnsiConsoleUtils`](https://github.com/mihnita/eclipse.platform.debug/blob/master/org.eclipse.ui.console/src/org/eclipse/ui/internal/console/ansi/AnsiConsoleUtils.java)
> 
> ```
> AnsiConsoleUtils.ESCAPE_SEQUENCE_REGEX_TXT.matcher(plainText).replaceAll("");
> ```

Thanks for that hint. Unfortunately that class is not accessible because the package is not exported.

In general and as it has already been said by Mickael earlier in this PR, unescaping of the console document text should not be done in M2E because others are probably also need this capability (getting the unescaped text, adding a link based on the unescaped offsets and lengths, etc.). Therefore I think it is better handled in the `o.e.ui.console` or `o.e.debug.ui`.
I'm thinking about having something like un-escaped views of an `IRegion` that provides the text without escape characters. Maybe it is even better to add an `getUnescapedDocument()` to the `IConsole`.
Or simply just build the entire document model by default on the unescaped text. So that if one really wants the raw text, it has to be obtained via `getRawDocument()` or so.
But that is something we should discuss in a dedicated issue in eclipse.platform.debug.

I also noticed that if the output is colored links are not displayed. But that is also an issue for platform.debug.


For now to support colored print-out in M2E I think this is the way we have to go until platform provides a more general one.

Reply:
> BTW, I know why the external maven uses colors on the level and the embedded one does not.

What you describes makes totally sense to me, although I have not yet fully tested it but you are right with the jars in the external maven installation.

I'm in favor of option 2. The logging is a bit difficult here, because we want to have the standard logging when we run a Maven-build launched from the IDE, but at the same time want to use the logback implementation when maven runs in the IDE and that only be achieved the way it is now (at least nobody had a better idea how to make that `generally` better).

But option 2 looks like a very good way to go that is simple to achieve. Do you want to provide a PR for that?
You just have to replace the slf4j-simple by `org.apache.maven:maven-slf4j-provider` in the `org.eclipse.m2e.maven.runtime.slf4j.simple` and adjust the version to match the embedded maven version. For that you probably want to move the `maven-core.version` property up to the `m2e-maven-runtime/pom.xml`.

Actually we should then remove the `simple` suffix from `org.eclipse.m2e.maven.runtime.slf4j.simple` and set it's version also to the maven version. In the require-bundle in the `org.eclipse.m2e.maven.runtime/pom.xml` has to be adjusted to the names as well.


Option 3 and 4 seem to be a bit overkill respectively overprotective. We already rely on what Maven is giving us and therefore we can rely on the logger as well.

Reply:
Looks like the renaming is a bit more work. It is probably simpler to split this, if you want you can just replace the logger impl and I take care of all other adjustments in the maven-components. Alternatively I can do all at once and mention you as coauthor of the commit.

Reply:
What do you think about adding `org.apache.maven:maven-slf4j-provider` directly to `org.eclipse.m2e.maven.runtime` and removing `org.eclipse.m2e.maven.runtime.slf4j.simple` completely?
It really is part of maven, covered by the same license, same version, etc.

M.

Reply:
> What do you think about adding `org.apache.maven:maven-slf4j-provider` directly to `org.eclipse.m2e.maven.runtime` and removing `org.eclipse.m2e.maven.runtime.slf4j.simple` completely?

I already thought about that for a while. But until know there where technical difficulties that make it harder to merge the sfl4j-simple jar into m2e.maven.runtime than keeping it separate.
But I had a new idea that should make it simple. I'm currently trying it out and will let you know if that has worked.
If it works as I intend it, then it should be a simple change to use `maven-slf4j-provider`.

Reply:
https://github.com/eclipse-m2e/m2e-core/pull/875